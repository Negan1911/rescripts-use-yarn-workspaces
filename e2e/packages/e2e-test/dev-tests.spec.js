const path = require('path')
const puppeteer = require('puppeteer')
const { exec } = require('child_process')
const { createServer } = require('http-server')

const app = path.resolve(__dirname, '../e2e-app')

function CompileApp() {
  return new Promise((resolve, reject) => {
    exec('yarn build', { cwd: app }, (err, stdout, stderr) => {
      if (err) return reject(err)
      if (stderr.length) return reject(stderr)
      if (!stdout.includes('Compiled successfully.'))
        reject(stdout)
        
      resolve()
    })
  })
}

jest.setTimeout(300000)
describe('dev tests', () => {
  const server = createServer({ root: path.join(app, 'build') })

  beforeAll((done) => {
    server.listen(30000, () => done())
  })

  afterAll(() => {
    server.close()
  })

  it('runs the page properly', async () => {
    const browser = await puppeteer.launch({ headless: true })
    const page = await browser.newPage()
    await page.goto(`http://localhost:30000`)
    await page.waitForSelector('button')
    const el = await await page.evaluate(() => {
      return document.querySelector('button').textContent;
    });
    
    expect(el).toBe('A Button!')
    expect(await page.evaluate(() => document.querySelector('#root').outerHTML)).toMatchSnapshot()

    await browser.close()
    expect(true)
  })
})