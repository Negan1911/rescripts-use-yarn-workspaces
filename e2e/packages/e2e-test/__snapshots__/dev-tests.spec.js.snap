// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dev tests runs the page properly 1`] = `"<div id=\\"root\\"><div><h1>This example imports a Button from <code>e2e-ui</code> package:</h1><button class=\\"sc-AxjAm bcMPWx\\">A Button!</button></div></div>"`;
